File: 0049-group-anagrams/0049-group-anagrams.py
from collections import defaultdict
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dic=defaultdict(list)
        result=[]
        
        for i in strs:
            s=tuple(sorted(i))
            dic[s].append(i)
        
        for i in dic.values():
            result.append(i)

        return result

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> myMap;
        for(int i=0; i<strs.size();i++){
            string s=strs[i];
            sort(s.begin(),s.end());
            myMap[s].push_back(strs[i]);  
        }
        vector<vector<string>> ans;
        for(auto& entry:myMap){
            ans.push_back(entry.second);
        }
        return ans;
    }
};